lib/periodicChecks.js

```diff
@@ -0,0 +1,234 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Handler for checks to be carried out periodically.
+ */
+
+const {
+  getAllOpenPullRequests,
+  hasPullRequestBeenApproved,
+  getChangelogLabelFromPullRequest,
+  getProjectOwnerFromLabel,
+  doesPullRequestHaveChangesRequested,
+  isUserAMemberOfTheOrganisation,
+} = require('./utils');
+const mergeConflictModule = require('./checkMergeConflicts');
+const { teamLeads } = require('../userWhitelist.json');
+
+/**
+ *  This function checks if a pull request has merge conflicts.
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const hasMergeConflict = (pullRequest) => {
+  return pullRequest.mergeable === false;
+};
+
+/**
+ *  This function checks if a pull request has pending reviews.
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const hasPendingReviews = (pullRequest) => {
+  return pullRequest.requested_reviewers.length > 0;
+};
+
+/**
+ * This function pings and assigns all pending reviewers to a pull request.
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const pingAndAssignReviewers = async (context, pullRequest) => {
+  const reviewers = pullRequest.requested_reviewers.map(
+    (reviewer) => reviewer.login
+  );
+
+  context.github.issues.addAssignees(
+    context.repo({
+      issue_number: pullRequest.number,
+      assignees: reviewers,
+    })
+  );
+
+  context.github.issues.createComment(
+    context.repo({
+      issue_number: pullRequest.number,
+      body:
+        'Assigning @' +
+        reviewers.join(', @') +
+        ' for codeowner reviews. Thanks!',
+    })
+  );
+};
+
+/**
+ * This function assigns the PR author to a pull request.
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ * @param {string} comment
+ */
+const pingAndAssignPrAuthor = async (context, pullRequest, comment) => {
+  context.github.issues.addAssignees(
+    context.repo({
+      issue_number: pullRequest.number,
+      assignees: [pullRequest.user.login],
+    })
+  );
+  await context.github.issues.createComment(
+    context.repo({
+      issue_number: pullRequest.number,
+      body: comment,
+    })
+  );
+};
+
+/**
+ * This function handles the state when a pull request has been approved.
+ * 1. Check if the PR author has merging rights and assign them to the PR.
+ * 2. If the PR author does not have merging rights, assign the PR to the
+ *    project owner from the changelog label.
+ * 3. If the pull request does not have a changelog label, assign the pr
+ *    author asking them to ping one of the reviewers to merge the PR.
+ *
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const handleApprovedPR = async (context, pullRequest) => {
```

Comment:
This is exactly similar to the function in pr reviewed module - why not import that and use the function directly?